### PR TITLE
autotools.eclass: Introduce AT_NOEAUTOHEADER variable

### PR DESCRIPTION
**Pull request by Polynomial-C**
Sometimes packages get their headers being broken by autoheader (see
media-libs/freetype for example) or get unwanted defines added (see
media-libs/libsdl2). In that case we want to run eautoreconf without
call to eautoheader.

Bug: https://bugs.gentoo.org/764959

### CODE REVIEW

File: eclass/autotools.eclass [hunk 1]
```diff
@@ -136,6 +136,12 @@ unset _automake_atom _autoconf_atom
 # Additional options to pass to automake during
 # eautoreconf call.
 
+# @ECLASS-VARIABLE: AT_NOEAUTOHEADER
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Don't run eautoheader command if set to 'yes'; only used to workaround
```
Comment (Arfrever) [inline eclass/autotools.eclass:142]: That noun can be written as `workaround`, but verb is `work around`.

Comment (Polynomial-C) [reply eclass/autotools.eclass]: Thanks. Fixed.

File: eclass/autotools.eclass [hunk 2]
```diff
@@ -65,7 +65,7 @@ inherit libtool
 # Do NOT change this variable in your ebuilds!
 # If you want to force a newer minor version, you can specify the correct
 # WANT value by using a colon:  <PV>:<WANT_AUTOMAKE>
-_LATEST_AUTOMAKE=( 1.16.1:1.16 1.15.1:1.15 )
+_LATEST_AUTOMAKE=( 1.16.3:1.16 1.16.1:1.16 )
```
Comment (Arfrever) [inline eclass/autotools.eclass:68]: This does not make sense.
If you want that latest mean only `1.16` slot, then use:
```
_LATEST_AUTOMAKE=( 1.16.3:1.16 )
```

**Review by floppym — APPROVED** [review]

**Comment by Arfrever** [comment]
Also copyright 2021 :) .

**Pull request closed without merging**